paymentsheet/src/main/java/com/stripe/android/customersheet/CustomerSheetLoader.kt

```diff
@@ -38,64 +36,50 @@ internal interface CustomerSheetLoader {
     private val workContext: CoroutineContext
 ) : CustomerSheetLoader {
 
     @Inject constructor(
         @Named(IS_LIVE_MODE) isLiveModeProvider: () -> Boolean,
         googlePayRepositoryFactory: @JvmSuppressWildcards (GooglePayEnvironment) -> GooglePayRepository,
-        elementsSessionRepository: ElementsSessionRepository,
         isFinancialConnectionsAvailable: IsFinancialConnectionsAvailable,
         lpmRepository: LpmRepository,
         errorReporter: ErrorReporter,
         @IOContext workContext: CoroutineContext
     ) : this(
         isLiveModeProvider = isLiveModeProvider,
         googlePayRepositoryFactory = googlePayRepositoryFactory,
-        elementsSessionRepository = elementsSessionRepository,
         isFinancialConnectionsAvailable = isFinancialConnectionsAvailable,
         lpmRepository = lpmRepository,
-        customerAdapterProvider = CustomerSheetHacks.adapter,
+        initializationDataSourceProvider = CustomerSheetHacks.initializationDataSource,
         errorReporter = errorReporter,
         workContext = workContext,
     )
 
     override suspend fun load(
         configuration: CustomerSheet.Configuration
     ): Result<CustomerSheetState.Full> = workContext.runCatching {
-        val customerAdapter = retrieveCustomerAdapter().getOrThrow()
-
-        val elementsSession = retrieveElementsSession(
-            customerAdapter = customerAdapter,
-        ).getOrThrow()
+        val initializationDataSource = retrieveInitializationDataSource().getOrThrow()
+        val customerSheetSession = initializationDataSource.loadCustomerSheetSession().toResult().getOrThrow()
 
         val metadata = createPaymentMethodMetadata(
             configuration = configuration,
-            elementsSession = elementsSession,
+            customerSheetSession = customerSheetSession,
         )
 
-        loadPaymentMethods(
-            customerAdapter = customerAdapter,
+        createCustomerSheetState(
+            customerSheetSession = customerSheetSession,
+            metadata = metadata,
             configuration = configuration,
-            elementsSessionWithMetadata = ElementsSessionWithMetadata(
-                elementsSession = elementsSession,
-                metadata = metadata,
-            ),
-        ).onFailure {
-            errorReporter.report(
-                errorEvent = ErrorReporter.ExpectedErrorEvent.CUSTOMER_SHEET_PAYMENT_METHODS_LOAD_FAILURE,
-                stripeException = StripeException.create(it)
-            )
-        }.getOrThrow()
+        )
     }
 
-    private suspend fun retrieveCustomerAdapter(): Result<CustomerAdapter> {
-        return customerAdapterProvider.awaitAsResult(
+    private suspend fun retrieveInitializationDataSource(): Result<CustomerSheetInitializationDataSource> {
+        return initializationDataSourceProvider.awaitAsResult(
             timeout = 5.seconds,
             error = {
                 "Couldn't find an instance of CustomerAdapter. " +
```

Comment:
Do we need to update this error message? I think with customer session, this will be incorrect. Maybe we need to move this error message into the individual implementations so we can provide better error messages for users

Reply:
Yeah we will need to be update it but unsure where this message should live.

Reply:
Actually users don't even receive this error message, I think it's only seen internally since this translates to `Something went wrong`.